cli/join.go

```diff
@@ -26,7 +26,9 @@ func join(c *cli.Context) {
 		log.Fatal("--heartbeat should be an unsigned integer and greater than 0")
 	}
 
-	d, err := discovery.New(dflag, hb)
+	// FIXME Add TLS
+	tls := &discovery.TLS{TLSConfig: nil}
```

Comment:
when we remove the `type TLSConfig struct` we will be able to use 

`d, err := discovery.New(dflag, hb, nil)`